File: lib/models/resi.dart
class Resi {
  final String id;
  final String noResi;
  final String title;
  final String courier;
  final DateTime createdAt;

  Resi({
    required this.id,
    required this.noResi,
    required this.title,
    required this.courier,
    required this.createdAt,
  });

  factory Resi.fromJson(Map<String, dynamic> json) {
    return Resi(
      id: json['_id'] ?? '',
      noResi: json['noResi'] ?? '',
      title: json['title'] ?? '',
      courier: json['courier'] ?? '',
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'noResi': noResi,
      'title': title,
      'courier': courier,
    };
  }
}

File: Express JS/models/Resi.js
const mongoose = require('mongoose');

const resiSchema = new mongoose.Schema({
    noResi: {
        type: String,
        required: true
    },
    title: {
        type: String,
        required: true
    },
    courier: {
        type: String,
        required: true,
        enum: ['jne', 'jnt', 'pos', 'sicepat', 'anteraja', 'tiki']
    },
    createdAt: {
        type: Date,
        default: Date.now
    },
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    }
});

// Membuat compound index untuk noResi dan userId
// Ini akan memastikan kombinasi noResi dan userId unik
resiSchema.index({ noResi: 1, userId: 1 }, { unique: true });

module.exports = mongoose.model('Resi', resiSchema, 'cekresi');
